Makefile.am

```diff
@@ -0,0 +1,369 @@
+
+# 1 - sources directory - with python package sources
+# 2 - destination directory - python package will be installed into the
+#     `packages` subdirectory of this destination directory
+define build_python_bundle
+	cd $(1) && \
+	PYTHONPATH=$(2)/packages/ \
+	LC_ALL=C.utf8 \
+	$(PYTHON) setup.py install --install-lib /packages/ --root $(2)
+endef
+
+PYAGENTX_URI="https://github.com/ondrejmular/pyagentx/archive/v${PYAGENTX_VERSION}.tar.gz"
+
+stamps/download_pyagentx:
+if ENABLE_DOWNLOAD
+	if [ ! -f ${abs_top_builddir}/rpm/pyagentx-${PYAGENTX_VERSION}.tar.gz ]; then \
+		$(WGET) -q -O ${abs_top_builddir}/rpm/pyagentx-${PYAGENTX_VERSION}.tar.gz ${PYAGENTX_URI}; \
+	fi
+endif
+	touch $@
+
+if PIP_HAS_ISOLATION
+pipopts	= --no-build-isolation
+endif
+
+stamps/download_python_deps: rpm/requirements.txt stamps/download_pyagentx dev_src/requirements.txt
+if ENABLE_DOWNLOAD
+	$(PIP) download --disable-pip-version-check --no-deps $(pipopts) --dest rpm/ --no-binary :all: -r rpm/requirements.txt
+endif
+	touch $@
+
+stamps/download_python_devel_deps: dev_src/requirements.txt
+if DEV_TESTS
+	$(PIP) download --disable-pip-version-check --no-deps $(pipopts) --dest dev_src/ --no-binary :all: -r dev_src/requirements.txt
+endif
+	touch $@
+
+if ENABLE_DOWNLOAD
+stamps/untar_python_src: stamps/download_python_deps stamps/download_python_devel_deps
+else
+stamps/untar_python_src:
+endif
+	$(MKDIR_P) ${abs_top_builddir}/$(PCS_BUNDLED_DIR_LOCAL)/src/
+	src=`ls -1 ${abs_top_builddir}/rpm/*.tar.gz ${abs_top_srcdir}/rpm/*.tar.gz ${abs_top_builddir}/dev_src/*.tar.gz ${abs_top_srcdir}/dev_src/*.tar.gz 2>/dev/null || true | sort -u | grep -v pcs- || true` && \
+	for i in $$src; do \
+		$(TAR) xvz -C ${abs_top_builddir}/${PCS_BUNDLED_DIR_LOCAL}/src -f $$i; \
+	done;
+if NOT_BUILD_DATACLASSES
+	rm -rf ${abs_top_builddir}/${PCS_BUNDLED_DIR_LOCAL}/src/dataclasses-*
+endif
+if NOT_BUILD_DISTRO
+	rm -rf ${abs_top_builddir}/${PCS_BUNDLED_DIR_LOCAL}/src/distro-*
+endif
+	touch $@
+
+stamps/install_python_embedded_mods_local: stamps/untar_python_src
+if LOCAL_BUILD
+	for i in ${abs_top_builddir}/$(PCS_BUNDLED_DIR_LOCAL)/src/*; do \
+		$(call build_python_bundle,$$i,/${abs_top_builddir}/$(PCS_BUNDLED_DIR_LOCAL)); \
+	done;
+endif
+	touch $@
+
+install_python_embedded_mods:
+if LOCAL_BUILD
+	for i in ${abs_top_builddir}/$(PCS_BUNDLED_DIR_LOCAL)/src/*; do \
+		$(call build_python_bundle,$$i,$(or ${DESTDIR}, /)/$(LIB_DIR)/$(PCS_BUNDLED_DIR)); \
+	done;
+endif
+
+stamps/install_ruby_deps_local:
+if LOCAL_BUILD
+if INSTALL_EMBEDDED_GEMS
+	rm -rf .bundle Gemfile.lock
+	$(MKDIR_P) .bundle
+	echo '---' > .bundle/config
+	echo 'BUNDLE_DISABLE_SHARED_GEMS: "true"' >> .bundle/config
+	echo 'BUNDLE_PATH: "${PCSD_BUNDLED_DIR_ROOT_LOCAL}"' >> .bundle/config
+	echo 'BUNDLE_BUILD: \"$(ruby_LIBS)\"' >> .bundle/config
+if ENABLE_DOWNLOAD
+else
+	echo 'BUNDLE_ALLOW_OFFLINE_INSTALL: "true"' >> .bundle/config
+endif
+	$(BUNDLE)
+	rm -rf .bundle Gemfile.lock
+endif
+	touch $@
+endif
```

Comment:
Shouldn't `touch $@` be outside of all if blocks?

Reply:
It doesn´t really matter. it can be inside or outside. The position is purely cosmetic.